### PR TITLE
OSSM-4194 Make Changes to the host in Destination Rule v2.3

### PR DESCRIPTION
**Pull request by mayleighnmyers**
https://issues.redhat.com/browse/OSSM-4194

### CODE REVIEW

**Comment by jewertow** [comment]
You should apply this change to the same file, but in the overlay directory.

File: resources/helm/overlays/mesh-config/templates/enable-mesh-mtls.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ metadata:
     app: istio
     release: {{ .Release.Name }}
 spec:
-  host: "*.local"
+  host: "*{{ .Values.global.proxy.clusterDomain }}"
```
Comment (nak3) [inline resources/helm/overlays/mesh-config/templates/enable-mesh-mtls.yaml]:
```suggestion
  host: "*.{{ .Values.global.proxy.clusterDomain }}"
```

File: resources/helm/v2.3/mesh-config/templates/enable-mesh-mtls.yaml [hunk 2]
```diff
@@ -29,7 +29,7 @@ metadata:
     app: istio
     release: {{ .Release.Name }}
 spec:
-  host: "*.local"
+  host: "*{{ .Values.global.proxy.clusterDomain }}"
```
Comment (nak3) [inline resources/helm/v2.3/mesh-config/templates/enable-mesh-mtls.yaml]:
```suggestion
  host: "*.{{ .Values.global.proxy.clusterDomain }}"
```

File: resources/helm/overlays/mesh-config/templates/enable-mesh-mtls.yaml (hunk 1 above)
Comment (mayleighnmyers) [reply resources/helm/overlays/mesh-config/templates/enable-mesh-mtls.yaml]: Thank you! That must've slipped past me in this version

**Review by luksa — APPROVED** [review]

**Comment by luksa** [comment]
@mayleighnmyers gen-check will pass once https://github.com/maistra/istio-operator/pull/1279 is merged.

**Pull request merged**